numpy/core/src/multiarray/calculation.c

```diff
@@ -35,12 +35,13 @@ power_of_ten(int n)
 }
 
 NPY_NO_EXPORT PyObject *
-_PyArray_ArgMaxWithKeepdims(PyArrayObject *op,
-        int axis, PyArrayObject *out, int keepdims)
+_PyArray_ArgMinMaxCommon(PyArrayObject *op,
+        int axis, PyArrayObject *out, int keepdims,
+        npy_bool is_argmax)
```

Comment:
Maybe instead of a `npy_bool` use `const char *` and pass in `""argmax"" or `"argmin"` directly? Other than that LGTM

Reply:
Maybe not, since then you would need to `strcmp` to get the function pointer

Reply:
> then you would need to `strcmp` to get the function pointer

Yes. I was also interested to use, `"argmax"` and `"argmin"` as arguments, but then introducing a dependency on `strcmp` just for an internal API didn't seem much beneficial, so thought of just going with `npy_bool`.